drivers/xen/gnttab.c

```diff
@@ -382,13 +380,6 @@ static int gnttab_init(void)
 		__ASSERT(!rc, "add_to_physmap failed; status = %d\n", rc);
```

Comment:
This question is not related to these changes, but AFAIU this assert will work only in test builds: https://elixir.bootlin.com/zephyr/v4.0.0/source/subsys/debug/Kconfig#L249.

Is this expected, or should we use something else for assertions in any case?

Reply:
IIUC the purpose of asserts is Zephyr - it is expected to appear as critical error on test builds and completely disappear on target builds. Since we depend on hypervisor here, just `if()` will be more correct, assert is legacy taken from mini-os driver.